src/hpguppi_ata_blade_obs_rawdisk_thread.c

```diff
@@ -265,19 +266,25 @@ static void *run(hashpipe_thread_args_t * args)
           // free previous fdraws
           free(fdraws);
 
-          nbeams = 1;
+          nbeams = 0;
           hgeti4(datablock_header, "NBEAMS", &nbeams);
-          if(beam_blocksize % nbeams == 0){
-            hashpipe_warn(thread_name, "BLOCSIZE %d split per NBEAMS %d.", beam_blocksize, nbeams);
-            beam_blocksize /= nbeams;
+          if(nbeams>0){
+            if(beam_blocksize % nbeams == 0){
+              hashpipe_warn(thread_name, "BLOCSIZE %d split per NBEAMS %d.", beam_blocksize, nbeams);
+              beam_blocksize /= nbeams;
```

Comment:
More generically the variable should be `file_blocksize`, if you're open to making that change while we're here :)